symfit/core/fit.py

```diff
@@ -1803,7 +1692,7 @@ def r_squared(model, fit_result, data):
     SS_tot = np.sum([np.sum((y_i - y_bar)**2) for y_i, y_bar in zip(y_is, y_bars) if y_i is not None])
     return 1 - SS_res/SS_tot
 
-class ODEModel(CallableModel):
+class ODEModel(BaseGradientModel):
```

Comment:
*not* a gradient model!

Reply:
Not sure I understand this remark, you mean why does it not inherit from GradientModel? Because for ODEModels it is hard to estimate the jacobian analytically, since that will itself be a system of ODEs. It's much easier to inherit the finite differences from BaseGradientModel.

Reply:
No, vice versa. `ODEModel` should *not* inherit from `BaseGradientModel` *because* we don't have an analytical Jacobian. And finite difference is too expensive to use during the minimization.

Reply:
However, we can estimate the jacobian using finite differences for the calculation of the covariance matrix